tools/pubsys/src/aws/promote_ssm/mod.rs

```diff
@@ -216,8 +231,86 @@ pub(crate) async fn run(args: &Args, promote_args: &PromoteArgs) -> Result<()> {
     Ok(())
 }
 
+/// Read parameters in given file, add newly promoted parameters, and write combined parameters to
+/// the given file
+async fn write_rendered_parameters(
+    ssm_parameters_output: &PathBuf,
+    set_parameters: &HashMap<SsmKey, String>,
+    source_target_map: HashMap<&String, &String>,
+) -> Result<()> {
+    info!(
+        "Writing promoted SSM parameters to {}",
+        ssm_parameters_output.display()
+    );
+    let parsed_parameters = parse_expected_parameters(&ssm_parameters_output.to_owned())
+        .await
+        .context(error::ParseExistingSsmParametersSnafu {
+            path: ssm_parameters_output,
+        })?;
```

Comment:
Am I correct that this will fail if the file is empty or doesn't exist? Is it better to just write the new parameters in that case, rather than refusing to proceed?

Reply:
That is correct, but I don't know if we just want to write the new parameters. One reason it could fail is that the syntax in the file is incorrect, in which case I think it's better to alert the user and halt the program than completely overwrite the content of the file.

Reply:
We chatted offline and concluded that we could specially handle `Error::ReadExpectedParameterFile`.